[RUBY-3764] Update waste_exemptions_engine branch and improve EA Public Face Area data loading

jjromeo opened the pull request:
https://eaflood.atlassian.net/browse/RUBY-3764
- Update gemfile to version of engine with ffi-geos
- Changed `EaPublicFaceAreaDataLoadService` to skip features marked as seaward (keep the land based features). The feature zip file contains duplicates of some of the regions which include an offshore and onshore version of some areas, before they were getting arbitrarily replaced based on the order of the file

jjromeo committed:
[RUBY-3764] Update waste_exemptions_engine branch and improve EA Public Face Area data loading

- Changed Gemfile to use the feature branch `RUBY-3764-wex-charging-implement-geospatial-ea-area-lookup` of waste_exemptions_engine
- Updated Gemfile.lock with new revision and added `ffi-geos` dependency
- Enhanced `EaPublicFaceAreaDataLoadService` to skip features marked as seaward and fixed property access for `code` in feature data loading

jjromeo committed:
[RUBY-3764] Rubocop fix

Refactor EA Public Face Area data loading by extracting feature processing into a separate `process_feature` method for improved code readability and maintainability.

jjromeo committed:
[RUBY-3764] Update waste_exemptions_engine gem reference to latest commit on RUBY-3764 branch

jjromeo committed:
[RUBY-3764] Update waste_exemptions_engine gem to track main branch and bump phonelib to 0.10.11

Changed the `waste_exemptions_engine` gem source branch from the feature branch `RUBY-3764-wex-charging-implement-geospatial-ea-area-lookup` to `main`. Also updated the `phonelib` gem version from 0.10.10 to 0.10.11 in Gemfile.lock.

jjromeo committed:
Merge branch 'main' into RUBY-3764-wex-charging-implement-geospatial-ea-area-lookup

brujeo commented on app/services/ea_public_face_area_data_load_service.rb:
```diff
@@ -8,29 +8,32 @@ def run
+
+    if area.present?
+      area.update(attributes)
+      puts "Updated EA Public Face Area \"#{area.code}\" (#{area.name})" unless Rails.env.test? # rubocop:disable Rails/Output
```
Why do we need this output here?

jjromeo committed:
[RUBY-3764] Refactor EA Public Face Area data load service to return results

- Modified `EaPublicFaceAreaDataLoadService#run` to collect and return results of created or updated areas instead of printing directly.
- Updated `process_feature` method to append actions to a results array.
- Changed rake task `load_admin_areas` to print results based on returned data, improving separation of concerns and testability.

jjromeo committed:
Merge branch 'RUBY-3764-wex-charging-implement-geospatial-ea-area-lookup' of https://github.com/DEFRA/waste-exemptions-back-office into RUBY-3764-wex-charging-implement-geospatial-ea-area-lookup
